checker/src/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -202,29 +201,36 @@ protected void annotateImplicit(Tree tree, AnnotatedTypeMirror type, boolean use
         dependentTypes.handle(tree, type);
     }
 
-
-    @Override
-    public AnnotatedTypeMirror getDefaultedAnnotatedType(Tree varTree,
-            ExpressionTree valueTree) {
-        AnnotatedTypeMirror result = super.getDefaultedAnnotatedType(varTree, valueTree);
-        return handlePolyNull(result, valueTree);
-    }
-
     /**
-     * Replaces {@link PolyNull} with {@link Nullable} to be more permissive
-     * (because {@code type} is usually a left-hand side) if the org.checkerframework.dataflow
-     * analysis has determined that this is allowed soundly.
+     * For types of left-hand side of an assignment, this method replaces {@link PolyNull} or
+     * {@link PolyAll} with  {@link Nullable} if the org.checkerframework.dataflow analysis
+     * has determined that this is allowed soundly.
+     * For example:
+     *
+     * <pre> @PolyNull String foo(@PolyNull String param) {
+     *    if (param == null) {
+     *        //  @PolyNull is really @Nullable, so change
+     *        // the type of param to @Nullable.
+     *        param = null;
+     *    }
+     *    return param;
+     * }
+     * </pre>
+     *
+     * @param lhsType  Type to replace
+     * @param context Tree used to get dataflow value
+     * @return altered lhsType
      */
-    protected AnnotatedTypeMirror handlePolyNull(AnnotatedTypeMirror type,
+    protected AnnotatedTypeMirror replacePolyQualifier(AnnotatedTypeMirror lhsType,
```

Comment:
The method side-effects the parameter `lhsType` and also returns it.
I would change the method to return `void` and document the side-effect.